spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/cql/CassandraConnector.scala

```diff
@@ -72,10 +77,6 @@ class CassandraConnector(conf: CassandraConnectorConf)
   def openSession() = {
     val session = sessionCache.acquire(_config)
     try {
-      val allNodes = session.getCluster.getMetadata.getAllHosts.toSet
-      val myNodes = LocalNodeFirstLoadBalancingPolicy.nodesInTheSameDC(_config.hosts, allNodes).map(_.getAddress)
-      _config = _config.copy(hosts = myNodes)
```

Comment:
Having looked at this once again I noticed a problem. Now when you call `openSession()`, the hosts do not get copied to _config, because `hosts` would be never called.